app/services/onlinetesting/Phase2TestService.scala

```diff
@@ -95,6 +95,25 @@ trait Phase2TestService extends OnlineTestService with Phase2TestConcern with Sc
     }
   }
 
+  def verifyAccessCode(email: String, accessCode: String): Future[Option[String]] = {
+    (for {
+      userId <- cdRepository.findUserIdByEmail(email)
+      testGroupOpt <- phase2TestRepo.getTestGroupByUserId(userId)
+    } yield {
+      testGroupOpt.flatMap { testGroup =>
```

Comment:
We could bump this up to the comprehension to remove one level of nesting and explicit mapping?